sql/sql_table.cc

```diff
@@ -3657,6 +3657,18 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
     if (sql_field->vcol_info)
       sql_field->flags&= ~NOT_NULL_FLAG;
 
+    if (sql_field->real_field_type() == MYSQL_TYPE_BLOB &&
+        thd->lex->sql_command == SQLCOM_CREATE_TABLE)
+    {
+      if (!strcmp(sql_field->type_handler()->name().ptr(), "MYSQL_JSON"))
+      {
+        my_printf_error(ER_UNKNOWN_ERROR, "Error parsing MySQL JSON format,"
```

Comment:
I suggest a better error message.

"mysql_json" is only meant to be used for upgrading from MySQL to MariaDB. Tables can not be created with "mysql_json" datatype.

Alternatively, how about use ER_CANT_CREATE_TABLE?

Reply:
Agree, will apply this error, like
```
ERROR HY000: Cannot create table `test`.`t1`: Run mariadb-upgrade, to upgrade table with mysql_json type.
```